### PR TITLE
revert: remove all code related to `get_sent_messages_request_event`

### PR DESCRIPTION
**Pull request by Julieta11**
Remove all code related to the endpoint `/v1/friendships/{friendshipId}/request-events/messages` as we have decided to include the sent message in the friend request as part of the conversation and retrieve it from there directly.

### CODE REVIEW

File: src/routes/v1/friendships/types.rs [hunk 1]
```diff
@@ -24,20 +24,20 @@ pub struct FriendshipFriend {
 
 #[derive(Debug, Default, Serialize, Deserialize)]
 pub struct MessageRequestEventResponse {
-    pub friendships: Vec<MessageRequestEvent>,
+    pub messages_req_events: Vec<MessageRequestEvent>,
```
Comment (lauti7) [inline src/routes/v1/friendships/types.rs:27]: just a comment to let you know that you can do something like the example below when we want another name for the internet.
```rust
pub struct MessageRequestEventResponse {
 #[serde(rename = "messages_request_event")]
 pub messages_req_events: Vec<MessageRequestEvent>,
}
```

check out https://serde.rs/field-attrs.html

Comment (Julieta11) [reply src/routes/v1/friendships/types.rs:27]: Amazing! thanks 🫶🏼

**Review by agusaldasoro — COMMENTED** [review]
Does this change the payload of the API or is just internal?

**Comment by Julieta11** [comment]
> Does this change the payload of the API or is just internal?

@agusaldasoro 
This change affects the fields of the API's payload. Internally, we'll manage the payload in the following format:
```
{
  "messages_request_events": [
    {
      "friendship_id": "67e55044-10b1-426f-9247-bb680e5fe0c8",
      "acting_user": "a_user_id",
      "timestamp": 1675189670,
      "body": "Hola"
    }
  ]
}
```
And then we'll serialize the payload to this format:
```
{
  "messagesRequestEvents": [
    {
      "friendshipId": "67e55044-10b1-426f-9247-bb680e5fe0c8",
      "actingUser": "a_user_id",
      "timestamp": 1675189670,
      "body": "Hola"
    }
  ]
}
```
The `#[serde(rename = "messages_request_event")]`  is not strictly necessary, but I believe it adds clarity.

**Review by agusaldasoro — APPROVED** [review]

File: src/routes/v1/friendships/history.rs [hunk 2]
```diff
@@ -22,26 +24,36 @@ use crate::{
...
         .expect("to have a UserId")
         .clone();
 
+    // Retrieve users from friendship and verify permissions.
+    let users = get_users_friendship(app_data.clone(), *friendship_id).await;
+    if !users.is_empty() {
+        has_permission(logged_in_user.social_id.as_str(), &users[0], &users[1]);
+    } else {
+        return Err(FriendshipsError::CommonError(CommonError::BadRequest(
+            format!("You don't have permission to view the sent request messages for friendship {friendship_id}"),
```
Comment (lauti7) [inline src/routes/v1/friendships/history.rs:51]: is this error description ok? I mean you are returning that the client doesn't have permission but the `if else` statement is checking that the `users` `Vec` is not empty.

**Review by lauti7 — APPROVED** [review]
Just some comments but it looks good overall!

File: src/routes/v1/friendships/history.rs [hunk 3]
```diff
@@ -22,26 +24,36 @@ use crate::{
...
+        )));
+    }
```
Comment (agusaldasoro) [inline src/routes/v1/friendships/history.rs:53]:
```suggestion
    if (users.is_empty() || !has_permission(logged_in_user.social_id.as_str(), &users[0], &users[1])) {
        return Err(FriendshipsError::CommonError(CommonError::BadRequest(
            format!("You don't have permission to view the sent request messages for friendship {friendship_id}"),
        )));
    }
```

File: src/routes/v1/friendships/history.rs (hunk 2 above)
Comment (agusaldasoro) [reply src/routes/v1/friendships/history.rs:51]: nice catch, see suggestion below

File: src/routes/v1/friendships/history.rs [hunk 4]
```diff
@@ -22,26 +24,36 @@ use crate::{
...
+
     // Convert it to a timestamp type that can be understood by PostgreSQL.
-    let timestamp_from_naive = match NaiveDateTime::from_timestamp_opt(body.timestamp_from, 0) {
+    let timestamp_from_naive = match NaiveDateTime::from_timestamp_opt(param.timestamp_from, 0) {
```
Comment (agusaldasoro) [inline src/routes/v1/friendships/history.rs:56]: can we name the query params just `from` and `to`?

**Review by agusaldasoro — CHANGES_REQUESTED** [review]

File: src/routes/v1/friendships/history.rs (hunk 2 above)
Comment (Julieta11) [reply src/routes/v1/friendships/history.rs:51]: Hahah nope, does not make sense at all. great catch ✨

File: src/routes/v1/friendships/history.rs [hunk 5]
```diff
@@ -21,56 +21,35 @@ use crate::{
...
-    let timestamp_to_naive = match NaiveDateTime::from_timestamp_opt(body.timestamp_to, 0) {
-        Some(val) => val,
-        None => {
-            return Err(FriendshipsError::CommonError(CommonError::BadRequest(
-                "Failed to convert timestamp_to to NaiveDateTime".to_string(),
-            )))
-        }
-    };
+    // Verify permissions.
+    if !has_permission(logged_in_user.social_id.as_str(), &users[0], &users[1]) {
```
Comment (agusaldasoro) [inline src/routes/v1/friendships/history.rs:43]: what about sending the array of users as the second param here?

File: src/routes/v1/friendships/history.rs [hunk 6]
```diff
@@ -21,56 +21,35 @@ use crate::{
...
 
-    let timestamp_to_naive = match NaiveDateTime::from_timestamp_opt(body.timestamp_to, 0) {
-        Some(val) => val,
-        None => {
-            return Err(FriendshipsError::CommonError(CommonError::BadRequest(
-                "Failed to convert timestamp_to to NaiveDateTime".to_string(),
-            )))
-        }
-    };
+    // Verify permissions.
```
Comment (agusaldasoro) [inline src/routes/v1/friendships/history.rs:42]: Let's add a more meaningful comment here, something like
```suggestion
    // Verify that the logged in user is present in the friendship
```

**Review by agusaldasoro — APPROVED** [review]
great work, just a few comments

**Review by agusaldasoro — APPROVED** [review]

**Pull request merged**